File: ios-app/add_build60_files.rb
#!/usr/bin/env ruby
require 'xcodeproj'

project_path = '/Users/expo/Code/expo/ios-app/PTPerformance/PTPerformance.xcodeproj'
project = Xcodeproj::Project.open(project_path)

# Find the main target
target = project.targets.find { |t| t.name == 'PTPerformance' }

# Helper to get or create groups
def get_or_create_group(parent, name)
  parent[name] || parent.new_group(name)
end

# Navigate to groups (PTPerformance is the main group in this project)
views_group = project.main_group['Views']
therapist_group = get_or_create_group(views_group, 'Therapist')
program_editor_group = get_or_create_group(therapist_group, 'ProgramEditor')
utils_group = project.main_group['Utils'] || project.main_group.new_group('Utils')
viewmodels_group = project.main_group['ViewModels']
tests_group = project.main_group['Tests']
integration_tests_group = get_or_create_group(tests_group, 'Integration')

# Files to add (relative to Views, Utils, ViewModels, Tests groups)
new_files = [
  # Agent 3: ProgramEditor Views
  { path: 'Views/Therapist/ProgramEditor/ProgramEditorView.swift', group: program_editor_group, target: target },
  { path: 'Views/Therapist/ProgramEditor/EditPhaseView.swift', group: program_editor_group, target: target },
  { path: 'Views/Therapist/ProgramEditor/EditSessionView.swift', group: program_editor_group, target: target },
  { path: 'Views/Therapist/ProgramEditor/ExerciseEditorView.swift', group: program_editor_group, target: target },
  { path: 'Views/Therapist/ProgramEditor/ExercisePickerView.swift', group: program_editor_group, target: target },

  # Agent 4: Utility Views
  { path: 'Utils/LoadingStateView.swift', group: utils_group, target: target },
  { path: 'Utils/ErrorStateView.swift', group: utils_group, target: target },

  # Agent 4: ViewModel
  { path: 'ViewModels/SessionSummaryViewModel.swift', group: viewmodels_group, target: target },

  # Agent 2: Tests
  { path: 'Tests/Integration/ProgramCreationTests.swift', group: integration_tests_group, target: nil }
]

files_added = 0
files_skipped = 0

new_files.each do |file_info|
  file_path = file_info[:path]
  group = file_info[:group]
  target = file_info[:target]
  basename = File.basename(file_path)

  # Check if file already exists in project
  existing_file = group.files.find { |f| f.path == basename }

  if existing_file
    puts "⏭️  Skipped (already exists): #{basename}"
    files_skipped += 1
  else
    # Add file reference
    file_ref = group.new_file(file_path)

    # Add to target if specified
    if target
      target.add_file_references([file_ref])
    end

    puts "✅ Added: #{basename}"
    files_added += 1
  end
end

# Save the project
project.save

puts "\n" + "="*60
puts "BUILD 60 FILES INTEGRATION COMPLETE"
puts "="*60
puts "Files added: #{files_added}"
puts "Files skipped: #{files_skipped}"
puts "Total files processed: #{files_added + files_skipped}"
puts "="*60

File: ios-app/add_build60_files.py
#!/usr/bin/env python3
"""
Add Build 60 UX Polish files to Xcode project
"""
import uuid
import re
import sys

def generate_uuid():
    """Generate a 24-character hex ID like Xcode uses"""
    return uuid.uuid4().hex.upper()[:24]

def add_file_to_project(project_path, file_path, file_name, group_name):
    """Add a file reference and build file to Xcode project"""

    with open(project_path, 'r') as f:
        content = f.read()

    # Generate UUIDs for the file
    file_ref_id = generate_uuid()
    build_file_id = generate_uuid()

    print(f"Adding {file_name}:")
    print(f"  File Ref ID: {file_ref_id}")
    print(f"  Build File ID: {build_file_id}")

    # Check if file already exists in project
    if file_name in content:
        print(f"  ⚠️  {file_name} already in project")
        return content, None, None

    # 1. Add PBXBuildFile entry
    build_file_entry = f"\t\t{build_file_id} /* {file_name} in Sources */ = {{isa = PBXBuildFile; fileRef = {file_ref_id} /* {file_name} */; }};\n"

    # Find the PBXBuildFile section and add after first entry
    build_file_pattern = r'(/\* Begin PBXBuildFile section \*/\n)'
    match = re.search(build_file_pattern, content)
    if match:
        insert_pos = match.end()
        content = content[:insert_pos] + build_file_entry + content[insert_pos:]
        print(f"  ✅ Added PBXBuildFile entry")
    else:
        print(f"  ❌ Could not find PBXBuildFile section")
        return content, None, None

    # 2. Add PBXFileReference entry
    file_ref_entry = f'\t\t{file_ref_id} /* {file_name} */ = {{isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = {file_path}; sourceTree = "<group>"; }};\n'

    # Find the PBXFileReference section
    file_ref_pattern = r'(/\* Begin PBXFileReference section \*/\n)'
    match = re.search(file_ref_pattern, content)
    if match:
        insert_pos = match.end()
        content = content[:insert_pos] + file_ref_entry + content[insert_pos:]
        print(f"  ✅ Added PBXFileReference entry")
    else:
        print(f"  ❌ Could not find PBXFileReference section")
        return content, None, None

    return content, file_ref_id, build_file_id

def add_to_sources_build_phase(content, build_file_id, file_name):
    """Add file to Sources build phase for PTPerformance target"""

    # Find the Sources build phase for PTPerformance target
    pattern = r'(/\* Sources \*/ = \{[^}]+isa = PBXSourcesBuildPhase;[^}]+files = \(\n)'
    match = re.search(pattern, content, re.DOTALL)

    if match:
        insert_pos = match.end()
        build_phase_entry = f'\t\t\t\t{build_file_id} /* {file_name} in Sources */,\n'
        content = content[:insert_pos] + build_phase_entry + content[insert_pos:]
        print(f"  ✅ Added to Sources build phase")
        return content
    else:
        print(f"  ❌ Could not find Sources build phase")
        return content

def add_to_group(content, file_ref_id, file_name, group_name):
    """Add file reference to appropriate group"""

    # Find the group by name
    pattern = rf'(/\* {group_name} \*/ = \{{[^}}]+children = \(\n)'
    match = re.search(pattern, content, re.DOTALL)

    if match:
        insert_pos = match.end()
        group_entry = f'\t\t\t\t{file_ref_id} /* {file_name} */,\n'
        content = content[:insert_pos] + group_entry + content[insert_pos:]
        print(f"  ✅ Added to {group_name} group")
        return content
    else:
        print(f"  ❌ Could not find {group_name} group")
        return content

def main():
    project_path = '/Users/expo/Code/expo/ios-app/PTPerformance/PTPerformance.xcodeproj/project.pbxproj'

    # Build 60 files to add
    files_to_add = [
        {
            'file_path': 'Utils/LoadingStateView.swift',
            'file_name': 'LoadingStateView.swift',
            'group_name': 'Utils'
        },
        {
            'file_path': 'Utils/ErrorStateView.swift',
            'file_name': 'ErrorStateView.swift',
            'group_name': 'Utils'
        },
        {
            'file_path': 'ViewModels/SessionSummaryViewModel.swift',
            'file_name': 'SessionSummaryViewModel.swift',
            'group_name': 'ViewModels'
        }
    ]

    # Read project file once
    with open(project_path, 'r') as f:
        content = f.read()

    # Backup original
    with open(project_path + '.build60_backup', 'w') as f:
        f.write(content)
    print("✅ Created backup: project.pbxproj.build60_backup\n")

    # Add each file
    for file_info in files_to_add:
        content, file_ref_id, build_file_id = add_file_to_project(
            project_path,
            file_info['file_path'],
            file_info['file_name'],
            file_info['group_name']
        )

        if file_ref_id and build_file_id:
            # Add to Sources build phase
            content = add_to_sources_build_phase(content, build_file_id, file_info['file_name'])

            # Add to group
            content = add_to_group(content, file_ref_id, file_info['file_name'], file_info['group_name'])

        print()

    # Write updated project file
    with open(project_path, 'w') as f:
        f.write(content)

    print("✅ Updated project.pbxproj")
    print("✅ All Build 60 UX Polish files added to Xcode project!")

if __name__ == '__main__':
    main()
